### PR TITLE
Add OData v2 batch example for generic OData client

### PR DESCRIPTION
**Pull request by artemkovalyov**
## What Has Changed?

Add example of implementing Batch read request via generic Odata client

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/java/features/odata/generic-odata-client.mdx [hunk 1]
```diff
@@ -160,3 +160,49 @@ if( deltaToken!=null) {
...
+Let's look how you can work around that with generic OData client for SAP S/4HANA `YY1_QUANTITYCONTRACTTRACKI_CDS`service
+
+In the code snippet below we:
+
+1. Initialize generic OData Batch request with service path and OData protocol version.
+1. Create a Map establishing relation between Contract object and Read request.
+1. Build Read requests for contracts we want to fetch and add them to Batch request and store in Map of Read requests.
+1. Make a Batch request
+1. Collate response of the Batch request to Read requests stored in the Map
+
```
Comment (artemkovalyov) [inline docs/java/features/odata/generic-odata-client.mdx:176]:
```suggestion
## OData v2 Batch Read Request

Batch requests for OData v2 are not yet supported in the [type-safe client](./type-safe-client-odata-v2.mdx#batch-requests).
Let's look at how you can work around that with a generic OData client for SAP S/4HANA `YY1_QUANTITYCONTRACTTRACKI_CDS`service

In the code snippet below we:

1. Initialize generic OData Batch request with service path and OData protocol version.
1. Create a Map establishing the relation between the Contract object and the Read request.
1. Build Read requests for contracts we want to fetch and add them to Batch request and store them in Map of Read requests.
1. Make a Batch request
1. Collate response of the Batch request to Read requests stored in the Map

```

File: docs/java/features/odata/generic-odata-client.mdx [hunk 2]
```diff
@@ -160,3 +160,49 @@ if( deltaToken!=null) {
...
+ODataRequestResultMultipartGeneric batchResult = requestBatch.execute(httpClient);
+
+// extract information from batch response, by referring to the individual OData request references
+for( Map.Entry<Contract, ODataRequestRead> requestMapping : batchedRequests.entrySet() ) {
+    ODataRequestResultGeneric queryResult = batchResult.getResult(requestMapping.getValue());
+    List<Map<String, Object>> itemsForQuery = queryResult.asListOfMaps();
+}
+```
+
+This example is a great demonstration how you can workaround certain limitation of type-safe client aka VDM, tackle advance OData features, or even take care of misbehaving servers or OData standard deviations.
```
Comment (artemkovalyov) [inline docs/java/features/odata/generic-odata-client.mdx:208]:
```suggestion
This example is a great demonstration of how you can work around certain limitations of a type-safe client aka VDM, tackle advanced OData features, or even take care of misbehaving servers or OData standard deviations.
```

File: docs/java/features/odata/generic-odata-client.mdx [hunk 3]
```diff
@@ -160,3 +160,49 @@ if( deltaToken!=null) {
     deltaRequest.addQueryParameter("$deltatoken", deltaToken);
 }
 ```
+
+## OData v2 Batch Read Request
+
+Batch requests for OData v2 are not yet supported in the [type-safe client](./type-safe-client-odata-v2.mdx#batch-requests).
+Let's look at how you can work around that with a generic OData client for SAP S/4HANA `YY1_QUANTITYCONTRACTTRACKI_CDS`service
```
Comment (artemkovalyov) [inline docs/java/features/odata/generic-odata-client.mdx:167]:
```suggestion
Let's look at how you can work around that with a generic OData client.
```

File: docs/java/features/odata/generic-odata-client.mdx [hunk 4]
```diff
@@ -160,3 +160,49 @@ if( deltaToken!=null) {
...
+In the code snippet below we:
+
+1. Initialize generic OData Batch request with service path and OData protocol version.
+1. Create a Map establishing the relation between the Contract object and the Read request.
+1. Build Read requests for contracts we want to fetch and add them to Batch request and store them in Map of Read requests.
+1. Make a Batch request
+1. Collate response of the Batch request to Read requests stored in the Map
+
+```java
+String servicePath = "/sap/opu/odata/sap/YY1_QUANTITYCONTRACTTRACKI_CDS";
```
Comment (artemkovalyov) [inline docs/java/features/odata/generic-odata-client.mdx:178]:
```suggestion
String servicePath = "/sap/opu/odata/sap/<SERVICE_PATH>";
```

File: docs/java/features/odata/generic-odata-client.mdx [hunk 5]
```diff
@@ -160,3 +160,49 @@ if( deltaToken!=null) {
...
+}
+
+// execute the OData batch request
+ODataRequestResultMultipartGeneric batchResult = requestBatch.execute(httpClient);
+
+// extract information from batch response, by referring to the individual OData request references
+for( Map.Entry<Contract, ODataRequestRead> requestMapping : batchedRequests.entrySet() ) {
+    ODataRequestResultGeneric queryResult = batchResult.getResult(requestMapping.getValue());
+    List<Map<String, Object>> itemsForQuery = queryResult.asListOfMaps();
+}
```
Comment (newtork) [inline docs/java/features/odata/generic-odata-client.mdx:205]:
```suggestion
1. Prepare an instance of `HttpClient` that will be used to execute the HTTP request.
1. Initialize a generic OData Batch request with OData service path and protocol version.
1. Build a read request for an OData entity collection and add it to the Batch request object.
1. Execute the Batch request.
1. Access the Batch response and fetch the list of fields from OData entities, that matched the applied read request.

```java
HttpDestination destination;
String servicePath = "/sap/opu/odata/sap/<SERVICE_PATH>";
String entityName = "EntityCollectionName";
HttpClient httpClient = HttpClientAccessor.getHttpClient(destination);

// instantiate custom OData V2 batch request
ODataRequestBatch requestBatch = new ODataRequestBatch(servicePath, ODataProtocol.V2);

// instantiate custom OData V2 read request
StructuredQuery structuredQuery = StructuredQuery.onEntity(entityName, ODataProtocol.V2);
structuredQuery.select("Field1", "Field2", "Field3");
structuredQuery.filter(FieldReference.of("Field4").equalTo("FooBar"));
String encodedQuery = structuredQuery.getEncodedQueryString();
ODataRequestRead requestRead = new ODataRequestRead(servicePath , entityName, encodedQuery, ODataProtocol.V2);

// add read request to batch
requestBatch.addRead(requestRead);

// execute the batch request
ODataRequestResultMultipartGeneric batchResult = requestBatch.execute(httpClient);

// extract information from batch response, by referring to the individual OData request reference
ODataRequestResultGeneric queryResult = batchResult.getResult(requestRead);
List<Map<String, Object>> listOfEntityFields = queryResult.asListOfMaps();
```

File: docs/java/features/odata/generic-odata-client.mdx [hunk 6]
```diff
@@ -160,3 +160,45 @@ if( deltaToken!=null) {
...
+
+// instantiate custom OData V2 batch request
+ODataRequestBatch requestBatch = new ODataRequestBatch(servicePath, ODataProtocol.V2);
+
+// instantiate custom OData V2 read request
+StructuredQuery structuredQuery = StructuredQuery.onEntity(entityName, ODataProtocol.V2);
+structuredQuery.select("Field1", "Field2", "Field3");
+structuredQuery.filter(FieldReference.of("Field4").equalTo("FooBar"));
+String encodedQuery = structuredQuery.getEncodedQueryString();
+ODataRequestRead requestRead = new ODataRequestRead(servicePath , entityName, encodedQuery, ODataProtocol.V2);
```
Comment (newtork) [inline docs/java/features/odata/generic-odata-client.mdx:191]: Remove unnecessary space.
```suggestion
ODataRequestRead requestRead = new ODataRequestRead(servicePath, entityName, encodedQuery, ODataProtocol.V2);
```

**Review by newtork — APPROVED** [review]
Looks good to me.

**Pull request merged**